applications/openshift/controller/controller_rotate_kubelet_server_certs/rule.yml

```diff
@@ -55,6 +55,23 @@ ocil: |-
     The output should return <tt>true</tt>.
 {{%- endif %}}
 
+{{%- if product == "ocp4" %}}
+warnings:
+- general: |-
```

Comment:
I'm afraid this rule doesn't work in cluster because the `warnigns` clause at the bottom of the rule rewrites the warnings that are supposed to fetch the API object. This is a proposed fix:
```
diff --git a/applications/openshift/controller/controller_rotate_kubelet_server_certs/rule.yml b/applications/openshift/controller/controller_rotate_kubelet_>
index 7ea7785cb..fb2175fb8 100644
--- a/applications/openshift/controller/controller_rotate_kubelet_server_certs/rule.yml
+++ b/applications/openshift/controller/controller_rotate_kubelet_server_certs/rule.yml
@@ -59,6 +59,18 @@ ocil: |-
 warnings:
 - general: |-
     {{{ openshift_cluster_setting("/api/v1/namespaces/openshift-kube-controller-manager/configmaps/config") | indent(4) }}}
+- functionality: |-
+    This recommendation only applies if you let kubelets get their
+    certificates from the API Server. In case your certificates come from an
+    outside Certificate Authority/tool (e.g. Vault) then you need to take care
+    of rotation yourself.
+{{% else %}}
+warnings:
+- functionality: |-
+    This recommendation only applies if you let kubelets get their
+    certificates from the API Server. In case your certificates come from an
+    outside Certificate Authority/tool (e.g. Vault) then you need to take care
+    of rotation yourself.
 {{%- endif %}}

 references:
@@ -80,10 +92,3 @@ identifiers:

 references:
     cis: 1.3.6
-
-warnings:
-    - functionality: |-
-        This recommendation only applies if you let kubelets get their
-        certificates from the API Server. In case your certificates come from an
-        outside Certificate Authority/tool (e.g. Vault) then you need to take care
-        of rotation yourself.
```

Reply:
@jhrozek does the old warning also still apply?

Reply:
I believe it does.

Reply:
@jhrozek went ahead and fixed the warning conflicts. Ignore that it is for OCP4 only in the if statement. We are going to remove OCP3 content.